File: src/main/java/toast/pine/Events/PlayerLevelUpEvent.java
package toast.pine.Events;


import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public class PlayerLevelUpEvent extends Event {

    private final HandlerList handlers = new HandlerList();

    private final Player player;
    private final Integer level;
    private final Integer maxLevel;
    private final Integer experience;

    public PlayerLevelUpEvent(Player player, Integer level, Integer maxLevel, Integer experience) {
        this.player = player;
        this.level = level;
        this.maxLevel = maxLevel;
        this.experience = experience;
    }

    public Player getPlayer() {
        return player;
    }

    public Integer getLevel() {
        return level;
    }

    public Integer getMaxLevel() {
        return maxLevel;
    }

    public Integer getExperience() {
        return experience;
    }

    @Override
    public @NotNull HandlerList getHandlers() {
        return handlers;
    }
}

File: src/main/java/pine/toast/toastrpg/library/events/PlayerLevelUpEvent.kt
package pine.toast.toastrpg.library.events

import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PlayerLevelUpEvent(thePlayer: Player, theLevel: Int, theMaxLevel: Int, theExperience: Int) : Event() {
    private val handlers = HandlerList()

    private var player: Player = thePlayer
    private var level: Int = theLevel
    private var maxLevel: Int = theMaxLevel
    private var experience: Int = theExperience

    override fun getHandlers(): HandlerList {
        return handlers
    }

    fun getPlayer(): Player {
        return player
    }

    fun getLevel(): Int {
        return level
    }

    fun getMaxLevel(): Int {
        return maxLevel
    }

    fun getExperience(): Int {
        return experience
    }

    companion object {
        private val handlers = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList {
            return handlers
        }
    }
}
